selfie camera auto-adjustment [FIX]

magmukendi opened the pull request:
- Fixed the issue #32 
- Updated the screenshot module to v1.0.2

Here is a demo video
<img  width="700" src="https://github.com/JasperGame/zepeto-modules/assets/64711820/c0577600-ff43-4ef4-b681-9fb3afa6afca"></img>

Close #32

bohyunjung reviewed (commented):
@magmukendi

Before reviewing the logic, we must clarify whether the 'height' refers to the character scale or the height associated with ZEPETO Body type.

For now, comments are made for other parts of the request.

magmukendi commented on Project/zepeto-modules/Assets/Zepeto ScreenShot Module/ZepetoScript/ScreenShotModeManager.ts:
```diff
@@ -30,6 +31,7 @@ export default class ScreenShotModeManager extends ZepetoScriptBehaviour {
         ZepetoPlayers.instance.OnAddedLocalPlayer.AddListener(() => {
             this.localPlayer = ZepetoPlayers.instance.LocalPlayer.zepetoPlayer;
             this.zepetoCamera = ZepetoPlayers.instance.LocalPlayer.zepetoCamera.camera;
+            this.myCharacter = Object.FindObjectOfType<ZepetoCharacter>();
```
Nope, figured out it doesn't.

I made some changes on line 34 [here](https://github.com/JasperGame/zepeto-modules/blob/0c9d30bc1efded1e5acc52b4aabbc18f5a0e6dc7/Project/zepeto-modules/Assets/Zepeto%20ScreenShot%20Module/ZepetoScript/ScreenShotModeManager.ts#L34).

magmukendi commented on Project/zepeto-modules/Assets/Zepeto ScreenShot Module/ZepetoScript/SelfieCamera.ts:
```diff
@@ -125,12 +131,31 @@ export default class SelfieCamera extends ZepetoScriptBehaviour {
+        }
+        else{
+            console.log(`Zepeto character scale CANNOT BE ${this._myCharacterScaleX}`)
+        }
```
Should have ask myself that question before. 
Will fix that in the next commit.

magmukendi commented:
So, In this commit [0c9d30b](https://github.com/JasperGame/zepeto-modules/pull/36/commits/0c9d30bc1efded1e5acc52b4aabbc18f5a0e6dc7), I have:
- refactored the variable myCharacter
- removed the redundant if else statement

And in this commit [bb102cc](https://github.com/JasperGame/zepeto-modules/pull/36/commits/bb102cc62a59c928884263353ff4688371e644d3), I have:
- hidden the myCharacter variable from the inspector 
- created a function (`setCameraHeight()`) that calculate the camera height according to the character LeftEye position (using the human body bones). 
- used the camera height and initial distance to compute the new distance between the character and the camera.

magmukendi commented:
@bohyunjung,

There was something wrong with the AdjustSelfieCamera function in the `SelfieCamera.ts.`. I made some more changes, Here are demo videos of how it is working now,

https://github.com/JasperGame/zepeto-modules/assets/64711820/66d40de8-a880-48cb-b259-2a42bf614c97


https://github.com/JasperGame/zepeto-modules/assets/64711820/92b913be-e8b9-4cef-b093-e1b2410875ea

bohyunjung commented on Project/zepeto-modules/Assets/Zepeto ScreenShot Module/ZepetoScript/SelfieCamera.ts:
```diff
@@ -125,12 +129,38 @@ export default class SelfieCamera extends ZepetoScriptBehaviour {
+    }
+
+    // This function returns the camera height following the head position;
+    private getSelfieCameraHeight(localPlayer:ZepetoPlayer): number {
```
What do you think of clarifying that this returns a 'new', 'processed' value for the selfie camera height?

I suggest `calculateSelfieCameraHeight`.

magmukendi replied:
Oh yeah, this makes more sense. I'll update this as soon as I can.

bohyunjung reviewed (commented):
@magmukendi Looks like a good solution for selfie height adjustment.
I've posted my opinion on a method. Please check.

bohyunjung commented on Project/zepeto-modules/Assets/Zepeto ScreenShot Module/ZepetoScript/SelfieCamera.ts:
```diff
@@ -1,8 +1,10 @@
+import { Vector3, Transform, Mathf, Object, Time, Quaternion, HideFlags, GameObject, Input, Application, HumanBodyBones } from 'UnityEngine';
 import { EventSystem } from 'UnityEngine.EventSystems';
+import ScreenShotModeManager from './ScreenShotModeManager';
+~import { ZepetoPlayer } from 'ZEPETO.Character.Controller';
```
```suggestion
import { ZepetoPlayer } from 'ZEPETO.Character.Controller';
```

bohyunjung reviewed (approved):
👍
